cmd/trader/trader.go

```diff
@@ -34,146 +41,232 @@ func main() {
+	// Get the Dark Pools
+	pools := ocean.GetPools()
+
+	if *cancelRequest {
+		cancelOrder(pools, multi)
+	} else {
+		// Keep sending order fragment
+		for {
+			// Get orders details from Binance
+			resp, err := http.Get(fmt.Sprintf("https://api.binance.com/api/v1/depth?symbol=ETHBTC&limit=%v", *numberOfOrders))
 			if err != nil {
-				log.Fatal("fail to parse the price into a big int")
+				log.Fatal("fail to get data from binance")
 			}
 
-			amount, err := strconv.ParseFloat(j[1], 10)
-			amount = amount * 1000000000000
-			if err != nil {
-				log.Fatal("fail to parse the amount into a big int")
+			response, err := ioutil.ReadAll(resp.Body)
+			orderBook := new(OrderBook)
+			err = json.Unmarshal(response, orderBook)
+
+			// Generate order from the Binance data
+			sellOrders := make([]*order.Order, len(orderBook.Asks))
+			for i, j := range orderBook.Asks {
+				price, err := strconv.ParseFloat(j[0], 10)
+				price = price * 1000000000000
+				if err != nil {
+					log.Fatal("fail to parse the price into a big int")
+				}
+
+				amount, err := strconv.ParseFloat(j[1], 10)
+				amount = amount * 1000000000000
+				if err != nil {
+					log.Fatal("fail to parse the amount into a big int")
+				}
+				order := order.NewOrder(order.TypeLimit, order.ParitySell, time.Time{},
+					order.CurrencyCodeETH, order.CurrencyCodeBTC, big.NewInt(int64(price)), big.NewInt(int64(amount)),
+					big.NewInt(int64(amount)), big.NewInt(1))
+				sellOrders[i] = order
+			}
+
+			buyOrders := make([]*order.Order, len(orderBook.Bids))
+			//test cast for match
+			for i, j := range orderBook.Asks { //change asks/bids
+				price, err := strconv.ParseFloat(j[0], 10)
+				price = price * 1000000000000
+				if err != nil {
+					log.Fatal("fail to parse the price into a big int")
+				}
+
+				amount, err := strconv.ParseFloat(j[1], 10)
+				amount = amount * 1000000000000
+				if err != nil {
+					log.Fatal("fail to parse the amount into a big int")
+				}
+
+				order := order.NewOrder(order.TypeLimit, order.ParityBuy, time.Time{},
+					order.CurrencyCodeETH, order.CurrencyCodeBTC, big.NewInt(int64(price)), big.NewInt(int64(amount)),
+					big.NewInt(int64(amount)), big.NewInt(1))
+				buyOrders[i] = order
+			}
+
+			// For all orders generated
+			for _, orders := range [][]*order.Order{buyOrders, sellOrders} {
+				go func(orders []*order.Order) {
+					for _, ord := range orders {
+						//todo
+						if ord.Parity == order.ParityBuy {
+							log.Println("sending buy order :", base58.Encode(ord.ID))
+						} else {
+							log.Println("sending sell order :", base58.Encode(ord.ID))
+						}
+
+						// For every Dark Pool
+						for i := range pools {
+
+							// Split order into (Number of nodes in each pool) * 2/3 fragments
+							shares, err := ord.Split(int64(pools[i].Size()), int64(pools[i].Size()*2/3), Prime)
+							if err != nil {
+								continue
+							}
+
+							// Send the shares across all nodes within the Dark Pool
```

Comment:
We should break this out into it's own function. It will make it easier to migrate to a package, and reuse in other places.

Reply:
Thanks, will work on that now.